syntax="proto2";
package KVS;
option cc_generic_services = true;

message GetRequest {
    required int64 key = 1;
};

message PutRequest {
    required int64 key = 1;
    required string value = 2;
};

message DelRequest {
    required int64 key = 1;
};

message KVStoreResponse {
    required bool success = 1;
    optional int64 key = 2;
    optional string value = 3;
    optional string redirect = 4;
};

service KVStoreService {
    rpc get(GetRequest) returns (KVStoreResponse);
    rpc put(PutRequest) returns (KVStoreResponse);
    rpc del(DelRequest) returns (KVStoreResponse);
};
